pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -675,3 +684,50 @@ type DigitalOceanSDConfig struct {
 	// +optional
 	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
 }
+
+// Docker SD configurations allow retrieving scrape targets from Docker Engine hosts.
+// This SD discovers "containers" and will create a target for each network IP and
+// port the container is configured to expose.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config
+// +k8s:openapi-gen=true
+type DockerSDConfig struct {
+	// Address of the docker daemon
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Host string `json:"host"`
```

Comment:
What do you think about remove [this part of the code](https://github.com/prometheus-operator/prometheus-operator/pull/6421/files#diff-aca3d056c83da1ddc448c8c40960960d7ed00ef61f8d5f08a40e165564d74dcdR1046) in favor of a kubebuilder annotation: `+kubebuilder:validation:Pattern`?

Reply:
I thought about kubebuilder validation but I wasn't sure if we can have the right regex prometheus accepts since prometheus is using url.Parse https://github.com/prometheus/prometheus/blob/main/discovery/moby/docker.go#L110-L112

url.Parse looks to be abstracted way for a complex regex validation?

Reply:
I see!
Yes docker host could be could be like that: `unix:///var/run/docker.sock` or other kind of dns such as http://

Reply:
Would we need to do the validation here? Or should we let it be handled at the url.Parse level JP mentioned at the Promtheus level?

Reply:
I'm fine using this only on the url.Parse especially because the regex pattern could be very complex.